docs/source/examples/learning-depolarizing-noise.myst

```diff
@@ -0,0 +1,208 @@
+---
+jupytext:
+  text_representation:
+    extension: .myst
+    format_name: myst
+    format_version: 0.13
+    jupytext_version: 1.14.1
+kernelspec:
+  display_name: Python 3.8.13 64-bit ('venv_mitiq_learnrepspec')
+  language: python
+  name:  python3
+---
+
+# Learning quasiprobability representations with a depolarizing noise model
+
++++
+
+In this example, we demonstrate the workflow of learning quasiprobability representations of a `CNOT` gate with a depolarizing noise model, from Clifford circuit data. The depolarizing noise model is parameterized by the noise strength, `epsilon`. The resulting quasiprobability representations of the `CNOT` gate are then used to obtain an error-mitigated expectation value with Mitiq's {ref}`probabilistic error cancellation module <guide/pec/pec>`.
+
+```{code-cell} ipython3
+import numpy as np
+from matplotlib import pyplot as plt
+from cirq import (
+    CXPowGate,
+    DepolarizingChannel,
+    Rx,
+    Rz,
+    I,
+    X,
+    Y,
+    Z,
+    LineQubit,
+    Circuit,
+    ops,
+    unitary,
+    InsertStrategy,
+)
+from mitiq import Executor, Observable, PauliString
+from mitiq.interface.mitiq_cirq import compute_density_matrix
+from mitiq.cdr import generate_training_circuits
+from mitiq.cdr._testing import random_x_z_cnot_circuit
+from mitiq.pec import execute_with_pec
+from mitiq.pec.representations.depolarizing import (
+    represent_operation_with_local_depolarizing_noise,
+)
+from mitiq.pec.representations.learning import (
+    biased_noise_loss_function,
+    learn_depolarizing_noise_parameter,
+)
+```
+
+Create a simple Rx-Rz-CNOT circuit using `mitiq.cdr._testing.random_x_z_cnot_circuit`. Here we use an (optional) seed for reproducibility.
+
+```{code-cell} ipython3
+circuit = random_x_z_cnot_circuit(
+    LineQubit.range(2), n_moments=5, random_state=np.random.RandomState(1)
+)
+print(circuit)
+```
+
+In this example, we will learn the strength of depolarizing noise of the `CNOT` gate in the circuit.
+
+```{code-cell} ipython3
+CNOT_ops = list(circuit.findall_operations_with_gate_type(CXPowGate))
+index = CNOT_ops[0][0]
+op = CNOT_ops[0][1]
+operations_to_learn = [Circuit(op)]
+```
+
+Specify the observable of interest.
+
+```{code-cell} ipython3
+observable = Observable(PauliString("XZ"), PauliString("YY"))
+```
```

Comment:
Is there anything special about this observable, or was it just chosen at random? If random, can we simplify and go without an explicit observable?

Reply:
Observable has to be defined for `compute_density_matrix`

Reply:
Moved to codeblock computing and plotting loss function.

Reply:
I don't think that's true. The definition of [`compute_density_matrix`](https://github.com/unitaryfund/mitiq/blob/1eca124f764d7b29ef5b751cce729479cdcfb92f/mitiq/interface/mitiq_cirq/cirq_utils.py#L61-L65) doesn't take an observable variable, and in the definition of [`evaluate`](https://github.com/unitaryfund/mitiq/blob/1eca124f764d7b29ef5b751cce729479cdcfb92f/mitiq/executor/executor.py#L119-L125) is the observable variable is optional.

https://github.com/unitaryfund/mitiq/blob/1eca124f764d7b29ef5b751cce729479cdcfb92f/mitiq/interface/mitiq_cirq/cirq_utils.py#L61-L65
https://github.com/unitaryfund/mitiq/blob/1eca124f764d7b29ef5b751cce729479cdcfb92f/mitiq/executor/executor.py#L119-L125

Reply:
Strange, I remember getting an error when setting the `Observable` to `None` when using `evaluate` with `compute_density_matrix`. Anyway, the specified observable happens to work well for the optimization. I've moved it into the code cell for computing the loss function, since I don't think it needs to be emphasized.